src/SmoFacade/Server.cs

```diff
@@ -224,7 +224,29 @@ public Database Database(string dbName)
 
         policy.Execute(() => restore.SqlRestore(_server));
         restore.Devices.Remove(backupDeviceItem);
+
+        if(fileRelocation != null)
+          LogicalFileRename(databaseName, fileRelocation);
+      }
+    }
+
+    private void LogicalFileRename(string databaseName, Func<string, string> fileRelocation)
+    {
+      var db = Database(databaseName);
+      if (db.Restoring)
+        db.RestoreWithRecovery();

```

Comment:
Don't want to have this logic in here... if you restore the database here, the rest of the chain won't be able to restore haha.

This function should only be called after the full chain in restored & finalized